### PR TITLE
Some New Features!

### PR DESCRIPTION
**Pull request by mojtaba1107**
Now after sending /config command, if user has no PV with bot or user has blocked bot, The bot will notice the user to Start first and this will reduce spams.

### CODE REVIEW

**Comment by mojtaba1107** [comment]
Now we have new achievement!!!
The Firefighter (as the GUARDIAN ANGEL clean At least two homes from kerosene of The arsonist.)

File: Werewolf for Telegram/Database/AchievementsReworked.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Database/AchievementsReworked.cs:203]: The casing here is kind of weird. 
```suggestion
        [Display(Name = "The Firefighter!"), Description("As the Guardian Angel, clean at least two homes from the Arsonist's kerosene.")]
```

File: Werewolf for Telegram/Shared/GameBalancing.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Shared/GameBalancing.cs:211]: Before, the new roles were always added at the end of the list, so you should stick to that.
The strength of 1 is arguable but since the Wolf Man has 1 too (why? 🤔 ) it's probably acceptable.

File: Werewolf for Telegram/Shared/Roles.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Shared/Roles.cs:145]: The 🌚 emoji is already used for the Wolf Man - it's better to avoid duplications like this.
And while the 🐐 emoji makes sense in English, it most likely does not in other languages.
How about this one, for example:
```suggestion
        [Role("😰")]
```

File: Werewolf for Telegram/Shared/Roles.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Shared/Roles.cs:146]:
```suggestion
        ScapeGoat = 17592186044416,
```

File: Werewolf for Telegram/Werewolf Node/Helpers/Settings.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Helpers/Settings.cs:108]: Really? This poor role has resistance against the cult?
Doesn't feel very logical to me...

File: Werewolf for Telegram/Werewolf Node/Helpers/Settings.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Helpers/Settings.cs:135]: Really? This poor role has resistance against the cult?
Doesn't feel very logical to me...

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:1544]: Again: New roles belong at the end of the list.

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:1647]: Unnecessary. This is taken care of by the default case in line 1700.

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:2747]: Local variable names should be `snakeCase` in C#
```suggestion
                            var scapeGoat = Players.FirstOrDefault(x => x.PlayerRole == IRole.ScapeGoat && !x.IsDead);
                            if (scapeGoat != null)
```

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:2750]: Logical error... At this point, `lynched` is the prince. What you're doing is saying that if there is a scape goat alive, the prince dies.
```suggestion
                                SendWithQueue(GetLocaleString("ScapeGoatLynched", scapeGoat.GetName()));
                                KillPlayer(scapeGoat, KillMthd.Lynch, killers: Players.Where(x => x.Choice == lynched.Id), isNight: false);
```
It is arguable who the scape goat's killers are in this case but the people who lynched the prince do make sense.

The other question is: Should the scape goat really die if the prince got lynched? The whole point in the prince role is that if the prince is lynched, the village realizes their mistake and refrains from lynching anyone that day.

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:2794]: Local variable names should be `snakeCase` in C#
```suggestion
                        var scapeGoat = Players.FirstOrDefault(x => x.PlayerRole == IRole.ScapeGoat && !x.IsDead);
                        if (scapeGoat != null)
```

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:2797]: At *this* point, `lynched` is an `IPlayer` object with no properties except for `Votes = -1` (see line 2690). Trying to get the name or kill them won't work.
```suggestion
                            SendWithQueue(GetLocaleString("ScapeGoatLynched", scapeGoat.GetName()));
                            KillPlayer(scapeGoat, KillMthd.Lynch, killers: Players.Where(x => x.Choice > 0), isNight: false);
```
Who are the killers in this case? Basically all people that voted to lynch someone.
Accessing `lynched.Id`, again, will not work because `lynched` is not an actual player at this point.

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:2804]: Local variable names should be `snakeCase` in C#
```suggestion
                        var scapeGoat = Players.FirstOrDefault(x => x.PlayerRole == IRole.ScapeGoat && !x.IsDead);
                        if (scapeGoat != null)
```

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:2807]: At *this* point, lynched is an IPlayer object with no properties except for Votes = -2 (see line 2694). Trying to get the name or kill them won't work.
```suggestion
                            SendWithQueue(GetLocaleString("ScapeGoatLynched", scapeGoat.GetName()));
                            KillPlayer(scapeGoat, KillMthd.Lynch, isNight: false);
```
Who are the killers in this case? Basically noone.
Accessing `lynched.Id`, again, will not work because `lynched` is not an actual player at this point.

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:3748]: Really? This poor role has resistance against the cult?
Doesn't feel very logical to me...
At least you added it to the end of the list this time :)

File: Werewolf for Telegram/Werewolf Node/Werewolf.cs
Comment (Olgabrezel) [inline Werewolf for Telegram/Werewolf Node/Werewolf.cs:5834]: Can't the player get the achievement if they were guardian angel before but got stolen?
Also, why do we need to wait until the game end to give out the achievement?
I'd suggest giving it mid-game below line 4065.

And *if* the new achievement is given at the game end, it should go at the end of the list!

**Review by Olgabrezel — CHANGES_REQUESTED** [review]
Holy moly.
This is 3 pull requests in one: A new achievement, a new role and requesting to start the bot upon `/config` command. You should consider making seperate pull requests for seperate things so we can merge those things we like and ignore those we don't.

For example: In the case that we do not want the ScapeGoat role, we can't merge this PR, even though the config start requesting might be useful.

The code is quite messy and partially has logic errors, I left more detailed review comments.
I put some *suggestions* but that doesn't necessarily mean that we will merge the pull request if you accept them all. However, it definitely cannot be merged in its current state.

Thanks a lot for the efford though, it is appreciated :)

**Pull request closed without merging**
